pyiceberg/io/pyarrow.py

```diff
@@ -414,19 +421,19 @@ def _initialize_oss_fs(self) -> FileSystem:
         return S3FileSystem(**client_kwargs)
 
     def _initialize_s3_fs(self, netloc: Optional[str]) -> FileSystem:
-        from pyarrow.fs import S3FileSystem, resolve_s3_region
+        from pyarrow.fs import S3FileSystem
 
         # Resolve region from netloc(bucket), fallback to user-provided region
         provided_region = get_first_property_value(self.properties, S3_REGION, AWS_REGION)
 
         try:
-            bucket_region = resolve_s3_region(bucket=netloc)
+            bucket_region = _cached_resolve_s3_region(bucket=netloc)
         except (OSError, TypeError):
             bucket_region = None
             logger.warning(f"Unable to resolve region for bucket {netloc}, using default region {provided_region}")
```

Comment:
How about moving try/except to `_cached_resolve_s3_region` and caching even the None value? Does the result change when we retry to resolve the same netloc? Or do we need to warn the user every time it fails to resolve the netloc?

Reply:
Hey @hussein-awala Thanks for jumping in here, and I really like your suggestion. Just pushed the suggestion, LMKWYT.

> Or do we need to warn the user every time it fails to resolve the netloc?

I was surprised by hearing that folks are seeing multiple warnings, since Python [only warns once by design](https://docs.python.org/3/library/warnings.html):

> Repetitions of a particular warning for the same source location are typically suppressed.